lib/kafka/connection.rb

```diff
@@ -89,6 +102,12 @@ def request(api_key, request, response_class)
           end
         end
       end
+    rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT => e
+      @logger.error "Connection error: #{e}"
+
+      close
```

Comment:
Retries and backoff is handled here: https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka/producer.rb#L182-L186